Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
With a "one connection per request" model there's no way of getting around needing a reference to the request. That would have to come in through the query API and would change public API for all of Fluent. 
It would have to be something like 
```swift
let users = User.query(request: req).all()
```